plugins/runner/internal/runtimehandler/binary/handler.go

```diff
@@ -0,0 +1,233 @@
+	if err != nil {
+		return fmt.Errorf("unable to remount input directory as read-only (%s - %s): %w", h.config.InputDir, h.inputDirMountPoint, err)
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			syscall.Unmount(h.inputDirMountPoint, 0)
+		}
+	}()
+
+	// Determine entrypoint or command to execute
+	var args []string
+	if len(image.Metadata.Config.Config.Entrypoint) > 0 {
+		args = append(image.Metadata.Config.Config.Entrypoint[0:], image.Metadata.Config.Config.Cmd...)
+	} else if len(image.Metadata.Config.Config.Cmd) > 0 {
+		args = image.Metadata.Config.Config.Cmd[0:]
+	} else {
+		return fmt.Errorf("no entrypoint or command found in the config")
+	}
+
+	// Find a port
+	openPortListener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return fmt.Errorf("unable to find port")
+	}
+	port := openPortListener.Addr().(*net.TCPAddr).Port
+
+	h.pluginServerEndpoint = fmt.Sprintf("http://127.0.0.1:%d", port)
+
+	// Set environment variables
+	env := image.Metadata.Config.Config.Env
+	env = append(env, fmt.Sprintf("%s=%s", plugin.EnvListenAddress, h.pluginServerEndpoint))
+
+	// Set workdir
+	workDir := image.Metadata.Config.Config.WorkingDir
+	if workDir == "" {
+		workDir = "/"
+	}
+
+	// Initialize command
+	h.cmd = exec.CommandContext(ctx, args[0], args[1:]...)
+	h.cmd.Env = env
+	h.cmd.Dir = workDir
+
+	h.cmd.SysProcAttr = &syscall.SysProcAttr{
+		Chroot: h.pluginDir,
+	}
+
+	h.cmd.Stdin = &bytes.Buffer{}
+	h.stdoutPipe, err = h.cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("unable to open pipe for command stdout: %w", err)
+	}
+
+	h.stderrPipe, err = h.cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("unable to open pipe for command stderr: %w", err)
+	}
+
+	// Start command
+	openPortListener.Close()
+	err = h.cmd.Start()
+	if err != nil {
+		return fmt.Errorf("unable to start process: %w", err)
+	}
+
+	// Waiting for command to be finished in the background
+	go func() {
+		h.cmd.Wait()
+	}()
+
+	h.ready = true
+
+	return nil
+}
+
+func (h *binaryRuntimeHandler) Ready() (bool, error) {
+	if h.cmd == nil {
+		return false, fmt.Errorf("plugin process is not running")
+	}
+
+	return h.ready, nil
+}
+
+func (h *binaryRuntimeHandler) GetPluginServerEndpoint(ctx context.Context) (string, error) {
+	return h.pluginServerEndpoint, nil
+}
+
+func (h *binaryRuntimeHandler) Logs(ctx context.Context) (io.ReadCloser, error) {
+	if h.cmd == nil {
+		return nil, fmt.Errorf("plugin process is not running")
+	}
+
+	stdoutPipeReader, stdoutPipeWriter := io.Pipe()
+	stderrPipeReader, stderrPipeWriter := io.Pipe()
+
+	go func() {
+		io.Copy(stdoutPipeWriter, h.stdoutPipe)
+	}()
+
+	go func() {
+		io.Copy(stderrPipeWriter, h.stderrPipe)
+	}()
+
+	return io.NopCloser(io.MultiReader(stdoutPipeReader, stderrPipeReader)), nil
+}
+
+func (h *binaryRuntimeHandler) Result(ctx context.Context) (io.ReadCloser, error) {
+	f, err := os.Open(filepath.Join(h.pluginDir, runtimehandler.RemoteScanResultFileOverride))
```

Comment:
Right now this is not a problem because the plugins are running sequentially on the inputs, but if we will run them concurrently we will need to create separate result files to avoid the collision.